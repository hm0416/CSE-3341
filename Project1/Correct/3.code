program
	int x;
begin
	ref y;
	ref z;
	y = new;
	y = 10;
	z = ref y;
	output z;
end