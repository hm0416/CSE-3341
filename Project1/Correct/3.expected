PROGRAM
INT
ID[X]
COMMA
ID[Y]
COMMA
ID[XY]
SEMICOLON
ID[a]
LPAREN
ID[b]
COMMA
ID[c]
COMMA
ID[d]
RPAREN
BEGIN
ID[a]
ASSIGN
ID[a]
ADD
ID[b]
ADD
ID[c]
SEMICOLON
ENDFUNC
SEMICOLON
BEGIN
ID[X]
ASSIGN
CONST[0]
SEMICOLON
OUTPUT
ID[X]
SEMICOLON
END
